### ISSUE TITLE
hreflang support

### ISSUE THREAD

**Issue by WBerndorfer** [nvaccess/nvda#13365]
# NVDA ignores html attribute `hreflang`

## What the spec says:

> The hreflang attribute on a elements that create [hyperlinks](https://html.spec.whatwg.org/#hyperlink), if present, gives the language of the linked resource. It is purely advisory. The value must be a valid BCP 47 language tag. [[BCP47]](https://html.spec.whatwg.org/#refsBCP47) User agents must not consider this attribute authoritative — upon fetching the resource, user agents must use only language information associated with the resource to determine its language, not metadata included in the link to the resource.

[]https://html.spec.whatwg.org/#attr-hyperlink-hreflang

# What screen readers should do
If an `<a>`-element has the attribute `hreflang`, then announce the meaning of the value of the attribute.
e.g.: link french …

## Use Case
When I encounter a link to a page or document in a language, I don’t know or hardly understand: I get immediate information about the language and not only after clicking on the link.

## Benefits:
-	Users get the information when browsing/navigating through a page, not only after clicking on the link.
-	The attribute is often already used for slim and site-wide design for links to foreign language resources. Screen Readers could ride the technically available horse to provide the information about the link too.
-	Screen Readers provide information about meta-data in the preferred language of the user. If a form element is provided with the attribute *required*, I get the German message “erforderlich”. This could also be achieved for the value of the foreign language.
# Related resources
## Language of page / parts (WCAG SC 3.1.1 and 3.1.2)
The targeted resource itself has to be provided with lang-attributes for page and deviating language parts. (WCAG SC 3.1.1 and 3.1.2)

## Styling
CSS Authors can use the attribute to visually place an icon with the language flag 
`a[hreflang="en"] { … }`

## Why `hreflang` is ignored by SR?
-	It was simply overlooked and not on the screen.
-	The development community is English and rarely confronted with foreign languages. They just underestimate the problem for users with other language preferences.

**Comment by WBerndorfer** [comment]
Sorry, the title should be:
**hreflang support**
Didn't find a way to correct it after publishing.

**Comment by seanbudd** [comment]
Our understanding is that this is intended for search engines. As far we are aware browsers do not report this to users, and it would add noise to screenreader users or be misleading.

**Comment by WBerndorfer** [comment]
How could browsers report it to me, when my SR doesn’t?
The browser support is OK:
https://caniuse.com/mdn-html_elements_a_hreflang

SEO: AT developers should not decide according to google results, but to the html spec.

I will discuss the issue in mailing lists.

**Comment by seanbudd** [comment]
How does your browser report hreflang to you? 

Can you point to the part of the HTML spec which suggests browsers or screenreaders should report hreflang?

>The hreflang attribute on [a](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element) elements that create [hyperlinks](https://html.spec.whatwg.org/multipage/links.html#hyperlink), if present, gives the language of the linked resource. It is purely advisory. The value must be a valid BCP 47 language tag. [[BCP47]](https://html.spec.whatwg.org/multipage/references.html#refsBCP47) User agents must not consider this attribute authoritative — upon fetching the resource, user agents must use only language information associated with the resource to determine its language, not metadata included in the link to the resource.

From the [html spec](https://html.spec.whatwg.org/multipage/links.html#attr-hyperlink-hreflang).

**Comment by seanbudd** [comment]
Sorry @WBerndorfer I misunderstood your message. Sighted users do not get hreflang information reported to them when browsing the internet. The attribute is intended for use by web crawlers.

**Comment by WBerndorfer** [comment]
Where do you read that the attribute was intended for crawlers?  In Google search results? Search engines look on the lang-value of the resource and won’t care much about the hint in a link to the resource.

But even if crawlers care about hreflang and authors use the attribute thinking about SEO:

Screen readers have to convey information technically provided. The more hreflang is used for SEO the more this is a reason to supply the attribute by NVDA. 

In other words:
1. NVDA has to use the hint in hreflang according to UAAG.
2. Users benefit, wether the attribute was used as an info for SR or for SEO.
